client/state/posts/test/actions.js

```diff
@@ -153,4 +163,70 @@ describe( 'actions', () => {
 			} ).catch( done );
 		} );
 	} );
+
+	describe( '#requestSitePost()', () => {
+		before( () => {
+			nock( 'https://public-api.wordpress.com:443' )
+				.persist()
+				.get( '/rest/v1.1/sites/2916284/posts/413' )
+				.reply( 200, { ID: 413, title: 'Ribs & Chicken' } )
+				.get( '/rest/v1.1/sites/2916284/posts/420' )
+				.reply( 404, {
+					error: 'unknown_post',
+					message: 'Unknown post'
+				} );
+		} );
+
+		after( () => {
+			nock.cleanAll();
+		} );
+
+		it( 'should dispatch request action when thunk triggered', () => {
+			requestSitePost( 2916284, 413 )( spy );
+
+			expect( spy ).to.have.been.calledWith( {
+				type: POST_REQUEST,
+				siteId: 2916284,
+				postId: 413
+			} );
+		} );
+
+		it( 'should dispatch posts receive action when request completes', ( done ) => {
```

Comment:
general thing, but i'm pretty sure you can just return the promise from requestSitePost.then(...) instead of using the done callback. https://mochajs.org/#working-with-promises